### PR TITLE
Convert Smart Proxy Plugin PR testing to a pipeline

### PR DESCRIPTION
**Pull request by ekohl**

### CODE REVIEW

**Comment by ekohl** [comment]
Testing this on https://ci.theforeman.org/job/smart_proxy_dns_infoblox-pr-test/ now.

**Comment by ekohl** [comment]
So it needs some work: https://github.com/theforeman/smart_proxy_dns_infoblox/pull/30. It's good work, since it brings the gem up to date and provides proper reporting of failures, but still work.

File: theforeman.org/yaml/jobs/tests/smart-proxy-plugin-pr-test.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    dsl:
+      !include-raw:
+        - pipelines/lib/git.groovy{empty}
+        - pipelines/lib/rvm.groovy{empty}
+        - pipelines/test/smart-proxy.groovy{empty}
+
+- project:
+    name: smart-proxy-plugin-pull-request
+    plugin:
+      - ansible
```
Comment (ekohl) [inline theforeman.org/yaml/jobs/tests/smart-proxy-plugin-pr-test.yaml:22]: https://github.com/theforeman/smart_proxy_ansible/pull/68

File: theforeman.org/yaml/jobs/tests/smart-proxy-plugin-pr-test.yaml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      - dhcp_infoblox
+      - dhcp_remote_isc
+      - dns_infoblox
```
Comment (ekohl) [inline theforeman.org/yaml/jobs/tests/smart-proxy-plugin-pr-test.yaml:25]: https://github.com/theforeman/smart_proxy_dns_infoblox/pull/30

File: theforeman.org/yaml/jobs/tests/smart-proxy-plugin-pr-test.yaml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+      !include-raw:
+        - pipelines/lib/git.groovy{empty}
+        - pipelines/lib/rvm.groovy{empty}
+        - pipelines/test/smart-proxy.groovy{empty}
+
+- project:
+    name: smart-proxy-plugin-pull-request
+    plugin:
+      - ansible
+      - dhcp_infoblox
```
Comment (ekohl) [inline theforeman.org/yaml/jobs/tests/smart-proxy-plugin-pr-test.yaml:23]: https://github.com/theforeman/smart_proxy_dhcp_infoblox/pull/53

File: theforeman.org/yaml/jobs/tests/smart-proxy-plugin-pr-test.yaml [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+      - dhcp_remote_isc
```
Comment (ekohl) [inline theforeman.org/yaml/jobs/tests/smart-proxy-plugin-pr-test.yaml:24]: https://github.com/theforeman/smart_proxy_dhcp_remote_isc/pull/7

**Comment by ehelms** [comment]
Generally code looks good

**Comment by ekohl** [comment]
@ehelms some Smart Proxy plugins do have Rubocop, others don't. Right now I'm going over them and add them where needed, but I wonder if we should make the pipeline aware of non-Rubocop projects by looking if `.rubocop.yml` exists (or otherwise, like via some variable). Any thoughts?

File: theforeman.org/yaml/jobs/tests/smart-proxy-plugin-pr-test.yaml [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+      - dns_infoblox
+      - discovery
+      - dynflow
+      - monitoring
+      - omaha
+      - openscap
+      - pulp
```
Comment (ekohl) [inline theforeman.org/yaml/jobs/tests/smart-proxy-plugin-pr-test.yaml:31]: https://github.com/theforeman/smart_proxy_pulp/pull/35

**Comment by ehelms** [comment]
> @ehelms some Smart Proxy plugins do have Rubocop, others don't. Right now I'm going over them and add them where needed, but I wonder if we should make the pipeline aware of non-Rubocop projects by looking if `.rubocop.yml` exists (or otherwise, like via some variable). Any thoughts?

If that makes it more flexible, and avoids having to sit on this change waiting on projects then I am all for that method.

**Comment by ekohl** [comment]
For now I've updated it to only convert plugins which have been updated, allowing to move forward quicker.

**Review by evgeni — APPROVED** [review]

**Pull request merged**